### ISSUE TITLE
Dynamic dateRangeInput start / end "frozen" after bookmark restore

### ISSUE THREAD

**Issue by andrewsali** [rstudio/shiny#2351]
The following simple app shows that after a dynamic dateRangeInput is restored, the start / end values no longer get properly updated when the dateRangeInput is redrawn. To reproduce this issue:

1. Start the app and verify that _source_ indeed propagates to _target_
2. Bookmark the app
3. Restore the app from the bookmark and verify that the source selection no longer propagates to target.

```
library(shiny)

ui <- function() {
  fluidPage(
    
    dateRangeInput("date1",
                   "Source:",
                   min = Sys.Date()-10,
                   max = Sys.Date()),
    uiOutput("date2-ui"),
    bookmarkButton("my_bookmark")
  )
}

server <- function(input, output) {
  
  output[["date2-ui"]] <- renderUI({
    req(input$date1)
    dateRangeInput("date2","Target:",min=Sys.Date()-10,max=Sys.Date(),start=input$date1[1],end=input$date1[2])
  })
}

shinyApp(ui = ui, server = server,enableBookmarking = "server")
```

This was not yet an issue with Shiny 1.0.1, but materialized when upgrading to 1.2.0.

**Comment by alandipert** [comment]
Hi @andrewsali, thank you for sharing your findings.

I was able to reproduce this behavior with 1.2.0 and observed it has changed since 1.1.0.

**Comment by alandipert** [comment]
@jcheng5 I bisected this to 0bb53e8ca5b8e769b11a0ec37a5b5c66eceb9a54

**Comment by wch** [comment]
Fixed in the rc-v1.3.0 branch by #2360.
